src/slider/slider.js

```diff
@@ -0,0 +1,170 @@
+import {} from './constants';
+import {getOverrideObject} from '../helpers/overrides';
+
+class Slider extends React.Component<PropsT, StatelessStateT> {
+  static defaultProps = {
+    overrides: {},
+    onChange: () => {},
+    onBlur: () => {},
+    onFocus: () => {},
+    onMouseEnter: () => {},
+    onMouseLeave: () => {},
+    onMouseDown: () => {},
+    onMouseUp: () => {},
+    error: false,
+    autoFocus: false,
+    tabIndex: 0,
+  };
+
+  state = {
+    isThumbMoving: false,
+    currentThumb: -1,
+    value: this.props.value
+  };
+
+  constructor(props: PropsT) {
+    super(props);
+  }
+
+  onFocus = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onBlur = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onMouseEnter = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onMouseLeave = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onChange = (
+    e: SyntheticEvent<HTMLInputElement>,
+    type: ChangeActionT,
+  ) => {
+  };
+
+  componentDidMount() {
+    this.domNode = findDOMNode(this);
+    this.addDocumentMouseEvents();
+  }
+
+  componentWillUnmount() {
+    if (super.componentWillUnmount) super.componentWillUnmount();
+    this.removeDocumentEvents();
+  }
+
+  onMouseDown = (e, thumbIndex) => {
+    this.setState({
+      isThumbMoving: true,
+      currentThumb: thumbIndex,
+    });
+  };
+
+  onMouseUp = e => {
+    if (this.state.isThumbMoving) {
+      this.setState({
+        isThumbMoving: false,
+        currentThumb: -1,
+      });
+    }
+  };
+
+  onMouseMove = e => {
+    if (this.state.isThumbMoving) {
+      this.onMove(e.movementX);
+    }
+  };
+
+  addDocumentMouseEvents() {
+    this.onMouseMoveListener = document.addEventListener('mousemove', this.onMouseMove);
+    this.onMouseUpListener = document.addEventListener('mouseup', this.onMouseUp);
+  }
+
+  removeDocumentEvents() {
+    this.onMouseMoveListener && this.onMouseMoveListener.remove();
+    this.onMouseUpListener && this.onMouseUpListener.remove();
+  }
+
+  render() {
+    const {overrides, range} = this.props;
+    const {value, currentThumb} = this.state;
+    const {component: Root, props: rootProps} = getOverrideObject(
+      overrides.Root,
+      StyledRoot,
+    );
+    const {component: Axis, props: axisProps} = getOverrideObject(
+      overrides.Axis,
+      StyledAxis,
+    );
+    const {component: Thumb, props: thumbProps} = getOverrideObject(
+      overrides.Thumb,
+      StyledThumb,
+    );
+    const {component: Tick, props: tickProps} = getOverrideObject(
+      overrides.Tick,
+      StyledTick,
+    );
+    const {component: TickBar, props: tickBarProps} = getOverrideObject(
+      overrides.TickBar,
+      StyledTickBar,
+    );
+    const sharedProps = {
+      $value: value,
+      $range: range,
+      $max: range[range.length - 1],
+      $min: range[0],
+      $currentThumb: currentThumb,
+    };
+    return (
+      <Root {...rootProps} {...sharedProps}>
+        <Axis {...axisProps} {...sharedProps}>
```

Comment:
Most other sliders support clicking on the bar to set the value

![kapture 2018-10-09 at 10 02 42](https://user-images.githubusercontent.com/875591/46685664-ddbf9800-cbaa-11e8-9fd6-515fae9e75d5.gif)

Do we plan to support this?

Reply:
I have added this feature for only 1 thumb